### ISSUE TITLE
Strange grepping...

### ISSUE THREAD

**Issue by qoole** [guydavis/machinaris#838]
Hi,

I noticed the log grepping taking a long time (and a lot of CPU time) on one of my boxes so I went to investigate.

```
# time grep -v 'Cumulative cost' debug.log.1 debug.log | grep -v 'CompressorArg' | grep -B 15 'Farmed unfinished_block' > /dev/null

real    0m35.883s
user    0m35.367s
sys     0m0.817s

```

The grep line in question is at line 101 in log_parser.py:
https://github.com/guydavis/machinaris/blob/b0037c41a56a901fbf452b93c350290ca83b5312/api/commands/log_parser.py#L101

This seems very inefficient, the line reads every line of debug.log/debug.log.1 returns every line except ones containing 'Cumulative cost' then scans every one of those remaining lines and returns all of the lines that don't contain 'CompressorArg' and then search all of the remaining lines for 'Farmed unfinished_block' and return it and the 15 lines that preceed it.

This could be slightly optimised by joining the first two greps together:

```
# time grep -ve 'Cumulative cost' -ve 'CompressorArg' debug.log debug.log.1 | grep -B 15 'Farmed unfinished_block' > /dev/null

real    0m25.042s
user    0m24.288s
sys     0m0.584s
```

Or invert the search order to for a huge jump:
```
# time grep -B 100 'Farmed unfinished_block' debug.log debug.log.1 | grep -ve 'Cumulative cost' -ve 'CompressorArg'

real    0m1.009s
user    0m0.476s
sys     0m0.463s
```

**Comment by guydavis** [comment]
Thanks for the feedback.  I'll give this a try, and also I'll decrease the frequency of the log search.  Really appreciate the [contribution](https://github.com/guydavis/machinaris/wiki/Contributing), so please let me know if you spot any other areas that could be improved.

**Comment by guydavis** [comment]
Available now in the Machinaris [v0.8.7 release](https://github.com/guydavis/machinaris/releases/tag/0.8.7).
